v1/topdown/topdown_partial_test.go

```diff
@@ -4200,6 +4200,768 @@ func TestTopDownPartialEval(t *testing.T) {
+					default q := false
+
+					q if { false }
+				`},
+			wantQueries: []string{""}, // unconditionally true
+		},
+
+		{
+			note:  "default rule inlining, default rule (int), result value presence",
+			query: "data.test.q",
+			modules: []string{`package test
+					default q := 42
+
+					q := 1 if { false }
+				`},
+			wantQueries: []string{""}, // unconditionally true (42 != false)
+		},
+		{
+			note:  "default rule inlining, default rule (int), result value unification (different value)",
+			query: "data.test.q = 1",
+			modules: []string{`package test
+					default q := 42
+
+					q := 1 if { false }
+				`},
+			wantQueries: []string{}, // unconditionally false
+		},
+		{
+			note:  "default rule inlining, default rule (int), result value unification (same value)",
+			query: "data.test.q = 42",
+			modules: []string{`package test
+					default q := 42
+
+					q := 1 if { false }
+				`},
+			wantQueries: []string{""}, // unconditionally true
+		},
+		{
+			note:  "default rule inlining, default rule (int), result value comparison (different value)",
+			query: "data.test.q == 1",
+			modules: []string{`package test
+					default q := 42
+
+					q := 1 if { false }
+				`},
+			wantQueries: []string{}, // unconditionally false
+		},
+		{
+			note:  "default rule inlining, default rule (int), result value comparison (same value)",
+			query: "data.test.q == 42",
+			modules: []string{`package test
+					default q := 42
+
+					q := 1 if { false }
+				`},
+			wantQueries: []string{""}, // unconditionally true
+		},
+
+		{
+			note:  "default rule inlining, default rule (boolean false), unknowns in undefined rule, result value presence",
+			query: "data.test.q",
+			modules: []string{`package test
+					default q := false
+
+					q if {
+						input.x == 7
+						false 
+					}
+				`},
+			wantQueries: []string{}, // unconditionally false
+		},
+		{
+			note:  "default rule inlining, default rule (boolean false), unknowns in undefined rule, result value unification (true)",
+			query: "data.test.q = true",
+			modules: []string{`package test
+					default q := false
+
+					q if {
+						input.x == 7
+						false 
+					}
+				`},
+			wantQueries: []string{}, // unconditionally false
+		},
+		{
+			note: "default rule inlining, default rule (boolean false), unknowns in undefined rule, result value comparison (true)",
+			// Compiled query:
+			// __localq0__ = data.test.q; equal(__localq0__, true)
+			query: "data.test.q == true",
+			modules: []string{`package test
+					default q := false
+
+					q if {
+						input.x == 7
+						false 
+					}
+				`},
+			// at eval-time of data.test.q, we don't know that __localq0__ is later compared with 'true' and support generation is superfluous.
+			// Can possibly be optimized into a no-support state by updating the compiler for this condition.
+			// E.g. for scalar values, replace equality (==) with unification (=): 'data.test.q == true' -> 'data.test.q = true'
```

Comment:
Yup. I suppose alternatively, we could have a post-processing step on the support modules? This could _then_ be dropped... 🤔